Adding Perf Test GH Action

jtaleric opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

All changes must adhere to this template to make it easy for reviewers
and preserve rationale/history behind every change
-->

## 📑 Description
<!-- Add a brief description of the pr -->
WIP Adding Performance job 

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Additional Information for reviewers
<!--
What exactly did you change - you may also defer to information
contained in commit messages. At a bare minimum it's worth highlighting
which areas of the code were changed as it's easier to assign reviewers
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [ ] My code requires changes to the documentation
- [ ] if so, I have updated the documentation as required
- [ ] My code requires tests
- [ ] if so, I have added and/or updated the tests as required
- [ ] All the tests have passed in the CI <!-- If not leave a comment as to why the CI is red and if you need help understanding what's wrong -->

## How to verify it
<!--
Did you include unit tests? or end-to-end tests?
How can I manually verify that this patch achieves its objective
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a CI performance testing pipeline that provisions kind clusters, builds/loads images, runs kube-burner workloads, collects profiles/metrics, uploads artifacts, and posts performance reports.
  * Optional infra nodes with automated Prometheus install and node-targeted metrics collection; new CLI options to configure infra count and metrics endpoint.
  * New performance tooling: standardized metrics, local/remote metric endpoints, workload templates, and a report generator that can save or post reports.

* **Documentation**
  * Added performance tooling README and usage guidance.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jtaleric commented:
/perf-test node-density-cni

npinaeva commented on .github/workflows/performance-test.yml:
```diff
@@ -0,0 +1,276 @@
+    branches: [ master ]
+
+env:
+  K8S_VERSION: v1.34.3
```
I think this version doesn't exist, try using 1.34.0 as the other jobs?

jtaleric committed:
Adding kubelet-density-cni performance workload

To run performance tests we needed to modify the kind cluster setup.
Creating infra nodes and installing prometheus is required so we can get
metrics about the SUT.

Installing and running kube-burner is also required for this new
workload.

In a follow on PR we will collect pprofs from the test

Signed-off-by: jtalerico <joe.talerico@gmail.com>

jtaleric committed:
Attempt to capture and store pprof data

Needed to change the endpoint we listen on for pprof. It was listening
on the pod ip -- instead we should listen on localhost.

Signed-off-by: jtalerico <joe.talerico@gmail.com>

trozet commented on dist/images/ovnkube.sh:
```diff
@@ -1319,8 +1319,7 @@ ovn-master() {
 
-  ovnkube_master_metrics_bind_address="${metrics_endpoint_ip}:9409"
-  ovnkube_master_metrics_bind_address="${metrics_endpoint_ip}:${metrics_master_port}"
+  ovnkube_master_metrics_bind_address="127.0.0.1:${metrics_master_port}"
```
we cant change it here. Earlier in the file there is:
```
metrics_endpoint_ip=${K8S_NODE_IP:-0.0.0.0}
```

we should change this to something like:
```
metrics_endpoint_ip="${METRICS_IP:-${K8S_NODE_IP:-0.0.0.0}}"
```

then we can propagate the value from kind.sh -> templates -> ovnkube.sh. This is a hassle but it is the way things work today. So in kind.sh and kind-helm.sh you can expose a new argument like --metrics-ip, which sets a var METRICS_IP. Then when ./daemonset.sh is called you pass --metrics-ip=$METRICS_IP, then in the template files like dist/templates/ovnkube-single-node-zone.yaml.j2 you edit this file to render the env variable METRICS_IP, and finally when ovnkube.sh runs it will use this value.

jtaleric committed:
Addressing trozet feedback

- remove tmate
- templating of the metrics_ip
- update permissions for action

Signed-off-by: jtalerico <joe.talerico@gmail.com>
